sdk/search/Azure.Search.Documents/src/Options/SearchOptions.cs

```diff
@@ -223,10 +223,17 @@ private string QueryCaptionRaw
         /// <summary> Allows the user to set an upper bound on the amount of time it takes for semantic enrichment to finish processing before the request fails. </summary>
         private int? SemanticMaxWaitInMilliseconds
         {
-            get { return SemanticSearch?.MaxWait?.Milliseconds; }
+            get
+            {
+                if (SemanticSearch?.MaxWait != null)
+                {
+                    return (int)SemanticSearch.MaxWait?.TotalMilliseconds;
+                }
+                return null;
+            }
```

Comment:
Tip: you can simply this with a cast to `int?`: https://sharplab.io/#v2:C4LglgNgNAJiDUAfAAgJgIwFgBQOAqYAtgKYDKADgIYB2A/AATADO9AvPdQK4QQDcOYasAZg29ABSDhASma0AdHgD2wShACykCGCbEAxkuowmvIA